aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/AWSXRayRecorder.java

```diff
@@ -415,6 +415,14 @@ public Segment beginSegmentWithSampling(String name) {
                 segment.setRuleName(samplingResponse.getRuleName().get());
             }
 
+            return segment;
+        } else if (this.getSamplingStrategy().isForcedSamplingSupported()) {
+            Segment segment = beginSegment(name);
+            segment.setSampled(false);
```

Comment:
From what I understand of ForcedSampling is that it can be used to sample a segment which is un-sampled when created. Or is it only to keep the segment in context even though it is un-sampled?

Reply:
Forced sampling forces the creation of a regular instead of no-op segment when not sampled, which also places it in context. Then `forceSamplingOfCurrentSegment` can be used in conjunction to actually switch an unsampled segment to sampled.